gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDatasetTest.java

```diff
@@ -0,0 +1,192 @@
+    MockDestinationFileSystem mockDestinationFileSystem = new MockDestinationFileSystem();
+    mockDestinationFileSystem.addPath(METADATA_PATH);
+    mockDestinationFileSystem.addPath(MANIFEST_PATH);
+    mockDestinationFileSystem.addPath(MANIFEST_LIST_PATH);
+    mockDestinationFileSystem.addPath(MANIFEST_FILE_PATH1);
+    mockDestinationFileSystem.addPath(MANIFEST_FILE_PATH2);
+
+    mockFileSystemMethodCalls(fs, mockDestinationFileSystem.pathToFileStatus, test_qualified_path, test_uri_path);
+
+    Collection<CopyEntity> copyEntities = icebergDataset.generateCopyEntities(copyConfiguration);
+    verifyCopyEntities(copyEntities, expected);
+
+  }
+
+  private void verifyCopyEntities(Collection<CopyEntity> copyEntities, List<String> expected) {
+    List<String> actual = new ArrayList<>();
+    for (CopyEntity copyEntity : copyEntities) {
+      String json = copyEntity.toString();
+      JsonObject jsonObject = new Gson().fromJson(json, JsonObject.class);
+      JsonObject objectData =
+          jsonObject.getAsJsonObject("object-data").getAsJsonObject("origin").getAsJsonObject("object-data");
+      JsonObject pathObject = objectData.getAsJsonObject("path").getAsJsonObject("object-data").getAsJsonObject("uri");
+      String filepath = pathObject.getAsJsonPrimitive("object-data").getAsString();
+      actual.add(filepath);
+    }
+    Assert.assertEquals(actual.size(), expected.size());
+    Assert.assertEqualsNoOrder(actual.toArray(), expected.toArray());
+  }
+
+  private void mockFileSystemMethodCalls(FileSystem fs, Map<Path, FileStatus> pathToFileStatus, String qualifiedPath, String uriPath)
+      throws URISyntaxException, IOException {
+
+    Mockito.when(fs.getUri()).thenReturn(new URI(null, null, uriPath, null));
+    for (Map.Entry<Path, FileStatus> entry : pathToFileStatus.entrySet()) {
+      Path path = entry.getKey();
+      FileStatus fileStatus = entry.getValue();
+      Mockito.when(fs.getFileStatus(path)).thenReturn(fileStatus);
+      Mockito.when(fs.makeQualified(path)).thenReturn(new Path(qualifiedPath));
+    }
+  }
+
+  private static class MockedIcebergTable extends IcebergTable {
+
+    String metadataPath;
+    String manifestListPath;
+    List<IcebergSnapshotInfo.ManifestFileInfo> manifestFiles;
+
+    public MockedIcebergTable(String metadataPath, String manifestListPath, List<IcebergSnapshotInfo.ManifestFileInfo> manifestFiles) {
+      super(null);
+      this.metadataPath = metadataPath;
+      this.manifestListPath = manifestListPath;
+      this.manifestFiles = manifestFiles;
+    }
+
+    @Override
+    public IcebergSnapshotInfo getCurrentSnapshotInfo() {
+      Long snapshotId = 0L;
+      Instant timestamp  = Instant.ofEpochMilli(0L);
+      return new IcebergSnapshotInfo(snapshotId, timestamp, metadataPath, manifestListPath, manifestFiles);
+    }
+  }
+
+  private static class MockDestinationFileSystem {
+    Map<Path, FileStatus> pathToFileStatus;
```

Comment:
this isn't actually itself a mock, so the name is misleading.  while you could encapsulate the entirety of the mocking into this class, it's also fine not to...

when you don't go all the way, please just name it appropriately.  e.g. `DestinationFileSystemPaths`.

Reply:
yeah makes sense.... I have updated it as part of the latest commit